sources/bloodhound/src/lib.rs

```diff
@@ -0,0 +1,320 @@
+where
+    I: IntoIterator<Item = S>,
+    S: AsRef<OsStr>,
+{
+    let search_strs = [search_str];
+    look_for_strings_in_output(cmd, args, &search_strs)
+}
+
+/// Executes a command and checks if the given `search_strs` are in the output.
+pub fn look_for_strings_in_output<I, S>(cmd: &str, args: I, search_strs: &[&str]) -> Option<bool>
+where
+    I: IntoIterator<Item = S>,
+    S: AsRef<OsStr>,
+{
+    if let Ok(output) = Command::new(cmd).args(args).output() {
+        if output.status.success() {
+            let mut matched = 0;
+
+            let mp_output = String::from_utf8_lossy(&output.stdout).to_string();
+            for line in mp_output.lines() {
+                for search_str in search_strs {
+                    if line.contains(search_str) {
+                        matched += 1;
+                    }
+                }
+            }
+
+            Some(search_strs.len() <= matched)
+        } else {
+            None
+        }
+    } else {
+        None
+    }
+}
+
+/// Execute a command and check if the output contains all provided strings, getting a `CheckerResult` of its findings.
+///
+/// If all `strings_to_match` are found in the output, the `CheckerResult` returned will have a `CheckStatus::PASS`.
+///
+/// If one or more of `strings_to_match` are not found in the output, then even if some are found the `CheckerResult`
+/// returned will have a `CheckStatus::FAIL` and the `error` field will contain `unable_to_find_error` as its content.
+///
+/// If execution fails or the output cannot be read, returned status will be `CheckStatus::SKIP` indicating a manual
+/// check will need to be performed and the `error` field will contain the `unable_to_check_error` value.
+#[macro_export]
+macro_rules! check_output_contains {
+    ($cmd:expr, $args:expr, $strings_to_match:expr, $unable_to_check_error:expr, $unable_to_find_error:expr) => {{
+        let mut result = results::CheckerResult {
+            error: String::new(),
+            status: results::CheckStatus::SKIP,
+        };
+
+        if let Some(found) = look_for_strings_in_output($cmd, $args, $strings_to_match) {
+            if found {
+                result.status = results::CheckStatus::PASS;
+            } else {
+                result.error = $unable_to_find_error.to_string();
+                result.status = results::CheckStatus::FAIL;
+            }
+        } else {
+            result.error = $unable_to_check_error.to_string();
+        }
+
+        result
+    }};
+}
+
+#[cfg(test)]
+mod test_utils {
+    use std::io::Write;
+    use tempfile::NamedTempFile;
+
+    use super::*;
+
+    #[test]
+    fn test_string_in_file_found() {
+        let mut test_file = NamedTempFile::new().unwrap();
+        writeln!(test_file, "udf 139264 0 - Live 0xffffffffc05e1000").unwrap();
+        writeln!(test_file, "crc_itu_t 16384 1 udf, Live 0xffffffffc05dc000").unwrap();
+        writeln!(test_file, "configfs 57344 1 - Live 0xffffffffc0320000").unwrap();
```

Comment:
Nice tests!

nit: could do something like this to line up the file contents:
```suggestion
        writeln!(test_file, concat!(
            "udf 139264 0 - Live 0xffffffffc05e1000\n",
            "crc_itu_t 16384 1 udf, Live 0xffffffffc05dc000\n",
            "configfs 57344 1 - Live 0xffffffffc0320000\n")
        )).unwrap();
```